be/src/storage/rowset/dict_page.cpp

```diff
@@ -235,7 +235,10 @@ Status DictPageDecoder<Type>::next_batch(const SparseRange<>& range, Column* dst
         _dict_decoder->at_index(codewords[i], &value);
         numbers[i] = value;
     }
-    dst->append_numbers(numbers.data(), numbers.size() * SIZE_OF_TYPE);
+    size_t nappend = dst->append_numbers(numbers.data(), numbers.size() * SIZE_OF_TYPE);
+    if (UNLIKELY(nappend != numbers.size())) {
+        return Status::InternalError("append numbers failed");
```

Comment:
Better to give these two numbers in the error message, otherwise nothing we can know from the error message.